### PR TITLE
[REF] mail: `AttachmentImage`, review and simplify scss

### PR DESCRIPTION
**Pull request by stefanorigano**
Part of the overall v16 SCSS optimization/restyle, #2704984

task: #2731819

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/mail/static/src/components/attachment_image/attachment_image.xml [hunk 1]
```diff
@@ -3,7 +3,7 @@
     <t t-name="mail.AttachmentImage" owl="1">
         <div t-attf-class="{{ className }}" t-ref="root">
             <t t-if="attachmentImage">
-                <div class="o_AttachmentImage d-flex o-details-overlay position-relative"
+                <div class="o_AttachmentImage d-flex o-details-overlay position-relative m-1 flex-shrink-0"
```
Comment (seb-odoo) [inline addons/mail/static/src/components/attachment_image/attachment_image.xml:6]: Isn't `o-details-overlay` now unused?

File: addons/mail/static/src/components/attachment_image/attachment_image.xml [hunk 2]
```diff
@@ -13,17 +13,17 @@
...
                             <i class="fa fa-spin fa-spinner"/>
                         </div>
                     </t>
-                    <div class="o_AttachmentImage_imageOverlay d-flex flex-row justify-content-end position-absolute">
-                        <div class="o_AttachmentImage_actions d-flex flex-column">
+                    <div class="o_AttachmentImage_imageOverlay d-flex flex-row justify-content-end p-2 text-white opacity-0 opacity-100-hover">
+                        <div class="o_AttachmentImage_actions d-flex flex-column justify-content-between">
                             <t t-if="attachmentImage.attachment.isEditable">
-                                <div class="o_AttachmentImage_action o_AttachmentImage_actionUnlink text-center"
+                                <div class="o_AttachmentImage_action o_AttachmentImage_actionUnlink btn btn-dark opacity-75 opacity-100-hover"
```
Comment (seb-odoo) [inline addons/mail/static/src/components/attachment_image/attachment_image.xml:26]: I think `o_AttachmentImage_action` is now unused too?

Comment (seb-odoo) [inline addons/mail/static/src/components/attachment_image/attachment_image.xml:26]: Also this button is now too visible I think, make sure the design is ok with FP. For v15 he requested that we made it more subtle.

File: addons/mail/static/src/components/attachment_image/attachment_image.xml [hunk 3]
```diff
@@ -13,17 +13,17 @@
...
                     <t t-if="attachmentImage.attachment.isUploading">
-                        <div class="o_AttachmentImageUploading d-flex align-items-center justify-content-center position-absolute" title="Uploading">
+                        <div class="o_AttachmentImageUploading d-flex align-items-center justify-content-center" title="Uploading">
                             <i class="fa fa-spin fa-spinner"/>
                         </div>
                     </t>
-                    <div class="o_AttachmentImage_imageOverlay d-flex flex-row justify-content-end position-absolute">
-                        <div class="o_AttachmentImage_actions d-flex flex-column">
+                    <div class="o_AttachmentImage_imageOverlay d-flex flex-row justify-content-end p-2 text-white opacity-0 opacity-100-hover">
+                        <div class="o_AttachmentImage_actions d-flex flex-column justify-content-between">
```
Comment (seb-odoo) [inline addons/mail/static/src/components/attachment_image/attachment_image.xml:24]: Isn't this div useless actually? It's doing flex stuff but there is only max 1 child.

File: addons/mail/static/src/components/attachment_image/attachment_image.xml (hunk 2 above)
Comment (stefanorigano) [reply addons/mail/static/src/components/attachment_image/attachment_image.xml:26]: Ok, I'll remove 👍

File: addons/mail/static/src/components/attachment_image/attachment_image.xml (hunk 1 above)
Comment (stefanorigano) [reply addons/mail/static/src/components/attachment_image/attachment_image.xml:6]: Ok, I'll remove 👍

File: addons/mail/static/src/components/attachment_image/attachment_image.xml (hunk 3 above)
Comment (stefanorigano) [reply addons/mail/static/src/components/attachment_image/attachment_image.xml:24]: Definitively 👍

File: addons/mail/static/src/components/attachment_image/attachment_image.xml (hunk 2 above)
Comment (stefanorigano) [reply addons/mail/static/src/components/attachment_image/attachment_image.xml]: Approved, he thinks it's an improvement 👍

**Comment by stefanorigano** [comment]
@seb-odoo , ready for tech review, thanks

**Review by seb-odoo — APPROVED** [review]
@robodoo r+ rebase-ff

**Pull request closed without merging**